Update "parallels" driver library and make this driver built into minikube

legal90 opened the pull request:
Changes:

- Fixes #9493 . As we can see from `go.mod`, minikube is already shipped with parallels driver built-in. The requirement of having `docker-machine-driver-parallels` binary available on the user's host is not needed, since it's not actually used by minikube. 

- Reverts #9494 because it's the opposite (alternative) way of fixing the same issue.

- Fixes #9357. The updated dependency `github.com/Parallels/docker-machine-parallels/v2 v2.0.1` includes the fix for macOS 11 BigSur. Ref: https://github.com/Parallels/docker-machine-parallels/pull/96

## Before this PR:
Make sure the `docker-machine-driver-parallels` is not available:
```
$ rm /usr/local/bin/docker-machine-driver-parallels
$ which docker-machine-driver-parallels
<empty response>
```
Then minikube requires user to install it:
```
$ minikube start --driver parallels
😄  minikube v1.14.0 on Darwin 10.15.7
    ▪ KUBECONFIG=/Users/legal/.kube/config
✨  Using the parallels driver based on existing profile

🤷  Exiting due to PROVIDER_PARALLELS_NOT_FOUND: The 'parallels' provider was not found: exec: "docker-machine-driver-parallels": executable file not found in $PATH
💡  Suggestion: Install docker-machine-driver-parallels
📘  Documentation: https://minikube.sigs.k8s.io/docs/reference/drivers/parallels/
```

## After this PR:
Now the driver is built-in, so no additional action is required form user (as long as Parallels Desktop for Mac is installed and license is activated)
```
$ minikube start --driver parallels
😄  minikube v1.14.0 on Darwin 10.15.7
    ▪ KUBECONFIG=/Users/legal/.kube/config
✨  Using the parallels driver based on existing profile
👍  Starting control plane node minikube in cluster minikube
💾  Downloading Kubernetes v1.19.2 preload ...
    > preloaded-images-k8s-v6-v1.19.2-docker-overlay2-amd64.tar.lz4: 486.33 MiB
🏃  Updating the running parallels "minikube" VM ...
🐳  Preparing Kubernetes v1.19.2 on Docker 19.03.12 ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" by default
```


cc @sharifelgamal , @afbjorklund , @tstromberg

legal90 commented:
/assign @tstromberg

medyagh reviewed (approved):
thank you for fixing it, I have no way of testing this myself but I trust you have tested this.

medyagh reviewed (commented):
do you mind putting the PR description the before and after this PR output ?

legal90 commented:
@medyagh Thanks you! Yes, I can add more details and provide "before" and "after" behavior later today.

But since you've merged https://github.com/kubernetes/minikube/pull/9494 I will also need to rebase this PR. Could you please tell me how do you want `parallels` driver being called: 
1) via RPC (as a standalone binary) ?
2) OR as a built-in library (similar to how it's done for `virtualbox` now) ?

tstromberg commented:
I personally think that doing so built-in creates a better user experience, rather than forcing users to download a separate helper binary.

tstromberg reviewed (approved):
LGTM from my perspective.

legal90 commented:
@tstromberg @medyagh I rebased the branch, updated the PR title and description.

The idea is still the same - this PR makes `parallels` driver built-in, so no separate driver binary is required anymore (so that, it reverts #9494). 
Also, it updates the driver to the latest version, which fixes https://github.com/kubernetes/minikube/issues/9357

tstromberg commented:
Thank you!
